### DISCUSSION TITLE
How to reset some state vars (or all  of them)

### DISCUSSION THREAD

**Discussion by PasqualePuzio** [reflex-dev/reflex#1985] [Q&A]
We would like to reset some state vars (or all of them) when the current page is closed, or alternatively, before the page is loaded.

What's the recommended approach to do that?

If we update some state vars upon the on_load event, would we see the old state first and then the updated one?

**Comment by masenf** [comment]
From an event handler, like on_load

```python
class State(rx.State):

    @rx.event
    def on_load(self):
        self.reset()

app = rx.App()
app.add_page(lambda: rx.text("hello world"), on_load=State.on_load)
```

**Reply by iBuitron** [reply]
@masenf is a @rx.event? i should be call on on_mount?

**Reply by masenf** [reply]
yes, it's an `@rx.event`

i would recommend page `on_load` for this, because that's going to reliably run on navigation events and it will execute in-order before the is_hyrdated flag gets set to True.
